elobot.py

```diff
@@ -11,195 +11,225 @@
-        games_csv = msg[(msg.index('>') + 1):]
-        games = games_csv.replace(' ', '').split(',')
-
-        for game in games:
-            scores = game.split('-')
-            if len(scores) != 2:
-                continue
+            time.sleep(0.1)
+            self.heartbeat()
+            self.ensure_connected()
+
+            messages = self.slack_client.rtm_read()
+            for message in messages:
+                if 'user' in message and message.get('type', False) == 'message' and message.get('channel', False) == self.channel_id and message.get('text', False):
+                    print('Message received:\n{}'.format(message))
+
+                    text = message['text']
+                    user_handle = message['user']
+                    has_backdoor = BACKDOOR_ENABLED and text.startswith(BACKDOOR_PREFIX)
+
+                    if WINNER_REGEX.match(text):
+                        loser_handle, winner_score, loser_score = re.search(WINNER_REGEX, text).groups()
+                        self.winner(user_handle, loser_handle, int(winner_score), int(loser_score))
+
+                    elif CONFIRM_REGEX.match(text):
+                        match_id, = re.search(CONFIRM_REGEX, text).groups()
+                        self.confirm(user_handle, match_id, has_backdoor)
+
+                    elif CONFIRM_ALL_REGEX.match(text):
+                        self.confirm_all(user_handle)
+
+                    elif DELETE_REGEX.match(text):
+                        match_id, = re.search(DELETE_REGEX, text).groups()
+                        self.delete(user_handle, match_id, has_backdoor)
+
+                    elif LEADERBOARD_REGEX.match(text):
+                        self.print_leaderboard()
+
+                    elif UNCONFIRMED_REGEX.match(text):
+                        self.print_unconfirmed()
+
+    def get_match(self, match_id):
+        """Get a match or say an error and return None"""
+        match = Match.select(Match).where(Match.id == match_id).get()
+        if not match:
+            self.talk(f'No match #{match_id}!')
+        return match
+
+    def get_pending(self, match_id):
+        """Get a pending match or say an error and return None"""
+        match = self.get_match(match_id)
+        if not match:
+            return None
+        if not match.pending:
+            self.talk(f'Match #{match_id} is not pending!')
+            return None
+        return match
+
+    def winner(self, winner_handle, loser_handle, winner_score, loser_score):
+        match = Match.create(winner_handle=winner_handle, winner_score=winner_score, loser_handle=loser_handle, loser_score=loser_score)
+        self.talk_to(loser_handle, f'Type "Confirm {match.id}" to confirm the above match, or ignore it if it\'s incorrect')
+
+    def confirm_all(self, user_handle):
+        matches = list(Match.select(Match).where(Match.loser_handle == user_handle, Match.pending == True))
```

Comment:
this should be sorted by the date created, because the order matches are confirmed in matter

Reply:
`confirm_all`?